arch/common/gen_isr_tables.py

```diff
@@ -143,13 +147,55 @@ def write_source_file(fp, vt, swt, intlist):
         fp.write("\t{(void *)0x%x, (void *)0x%x},\n" % (param, func))
     fp.write("};\n")
 
+def get_symbols(obj):
+    for section in obj.iter_sections():
+        if isinstance(section, SymbolTableSection):
+            return {sym.name: sym.entry.st_value
+            for sym in section.iter_symbols()}
+
+    error("Could not find symbol table")
+
+def getindex(irq, irq_aggregator_pos):
+    for indx, val in enumerate(irq_aggregator_pos):
+        if irq == irq_aggregator_pos[indx]:
+            return indx
+    error("The index %d has no match. Recheck interrupt configuration" % indx)
+
 def main():
     parse_args()
 
+    with open(args.kernel, "rb") as fp:
+        kernel = ELFFile(fp)
+        syms = get_symbols(kernel)
+
+    if "CONFIG_MULTI_LEVEL_INTERRUPTS" in syms:
+        if "CONFIG_2ND_LEVEL_INTERRUPTS" in syms:
+            if "CONFIG_2ND_LVL_INTR_0_OFFSET" in syms:
+                list_2nd_lvl_offsets = [syms["CONFIG_2ND_LVL_INTR_0_OFFSET"]]
+            if "CONFIG_2ND_LVL_INTR_1_OFFSET" in syms:
+                list_2nd_lvl_offsets.append(syms["CONFIG_2ND_LVL_INTR_1_OFFSET"])
+            if "CONFIG_2ND_LVL_INTR_2_OFFSET" in syms:
+                list_2nd_lvl_offsets.append(syms["CONFIG_2ND_LVL_INTR_2_OFFSET"])
+            if "CONFIG_2ND_LVL_INTR_3_OFFSET" in syms:
```

Comment:
Having CONFIG_2ND_LEVEL_INTR values from 0 - 3 seems specific to a particular board.
Can we make this more generic somehow? You have CONFIG_NUM_2ND_LEVEL_AGGREGATORS but do not use it anywhere that I can see.

Reply:
This could also be made as a `for` loop:

```Python
level = 0
offsets = []
while True:
    sym = syms.get("CONFIG_2ND_LVL_INTR_%d_OFFSET" % level)
    if not sym:
        break

    offsets.append(sym)
    level += 1
```

Reply:
Yes, you are right, Andrew. This script was written keeping in mind a particular board. Will make it generic.
Thanks for suggestion, Leandro. I too have something in mind and will submit it.